cmake/kokkos_arch.cmake

```diff
@@ -440,3 +444,14 @@ FOREACH(Arch ${KOKKOS_ENABLED_ARCH_LIST})
   MESSAGE(STATUS " ${Arch}")
 ENDFOREACH()
 
+IF (KOKKOS_ENABLE_CUDA)
+  FIND_PACKAGE(CUDA)
+  IF (CUDA_FOUND)
+    ADD_EXECUTABLE(kokkos_cuda_version ${CMAKE_CURRENT_LIST_DIR}/compile_tests/cuda_test.cc)
+    TARGET_LINK_LIBRARIES(kokkos_cuda_version ${CUDA_LIBRARIES})
+    TARGET_INCLUDE_DIRECTORIES(kokkos_cuda_version PRIVATE ${KOKKOS_TOP_BUILD_DIR} ${KOKKOS_SOURCE_DIR}/core/src)
+  ELSE()
+    MESSAGE(SEND_ERROR "No CUDA is found. Validation cannot be performed")
+  ENDIF()
+  ADD_CUSTOM_TARGET(cuda_version COMMAND kokkos_cuda_version)
```

Comment:
The intent is the user would call `make cuda_version`?

Reply:
Correct. You would crash and say you must have a CUDA arch specified, but give directions on how to figure it out.